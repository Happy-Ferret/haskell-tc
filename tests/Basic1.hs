module Basic1 where

-- data Unit = Unit

-- id :: a -> a
-- id x = x

-- ap = id Unit

twice fn v = fn (fn v)
