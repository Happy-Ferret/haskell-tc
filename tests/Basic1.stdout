Bindings:
:: âˆ€ v3. (v3 â†’  v3) â†’  v3 â†’  v3
  [4mtwice fn v = fn (fn v)[0m
:: v3 â†’  v3
  twice [4mfn[0m v = fn (fn v)
:: v3
  twice fn [4mv[0m = fn (fn v)

Proofs:
